backend/ai_integration/services.py

```diff
@@ -0,0 +1,299 @@
+                } if user_taste else {},
+                'owned_book_ids': [str(book.id) for book in user_library]
+            }
+        }
+    
+    @staticmethod
+    def recommend_books_for_barter(requester, recipient, requested_book, limit: int = 3) -> List[str]:
+        """
+        AI를 사용하여 교환에 적합한 책들을 추천.
+        실제 모델 호출 예시 포함.
+        """
+        context_data = AIRecommendationService.get_barter_context_data(
+            requester, recipient, requested_book
+        )
+        # 실제 AI 모델 호출 예시
+        try:
+            from pipeline.recommender import BarterRecommender
+            from data.entities import BarterContext, Item, UserProfile, TradeRequest
+            # 컨텍스트 변환 (간단 예시)
+            items = {}
+            for book in context_data['requester']['available_books']:
+                items[book['id']] = Item(
+                    item_id=book['id'],
+                    owner_id=context_data['requester']['id'],
+                    title=book['title'],
+                    valuation=1.0,
+                    facets={
+                        'genre': ','.join(book['genres']),
+                        'author': ','.join(book['authors'])
+                    },
+                    metadata={}
+                )
+            profiles = {
+                context_data['requester']['id']: UserProfile(
+                    user_id=context_data['requester']['id'],
+                    display_name=context_data['requester']['username'],
+                    trust_score=0.8,
+                    reliability=0.9,
+                    preferences=context_data['recipient']['taste'],
+                ),
+                context_data['recipient']['id']: UserProfile(
+                    user_id=context_data['recipient']['id'],
+                    display_name=context_data['recipient']['username'],
+                    trust_score=0.8,
+                    reliability=0.9,
+                    preferences=context_data['recipient']['taste'],
+                )
+            }
+            requests = [
+                TradeRequest(
+                    user_id=context_data['recipient']['id'],
+                    desired_item_ids=[context_data['recipient']['requested_book']['id']],
+                    desired_facets={}
+                )
+            ]
+            context = BarterContext(
+                items=items,
+                profiles=profiles,
+                requests=requests
+            )
+            recommender = BarterRecommender()
+            recommendations = recommender.recommend(context, limit=limit)
+            return [rec.candidate.item_id for rec in recommendations]
+        except Exception as e:
+            # 임시: 랜덤으로 선택 (AI 모델 통합 실패 시)
+            available_books = BookCopy.objects.filter(
+                owner=requester,
+                is_for_barter=True,
+                trade_status="available"
+            ).order_by('?')[:limit]
+            return [str(book.id) for book in available_books]
```

Comment:
Log the exception using e


```suggestion
        except Exception as e:
            # 임시: 랜덤으로 선택 (AI 모델 통합 실패 시)
            available_books = BookCopy.objects.filter(
                owner=requester,
                is_for_barter=True,
                trade_status="available"
            ).order_by('?')[:limit]
            return [str(book.id) for book in available_books]
```